pkg/lib/config/testdata/parse_secret_tests.yaml

```diff
@@ -188,4 +188,24 @@ config:
   secrets:
     - key: bot_protection.provider
       data:
-        type: hcaptcha # not supported
\ No newline at end of file
+        type: hcaptcha # not supported
+---
+name: ldap/valid
+error: null
+config:
+  secrets:
+    - key: ldap
+      data:
+        name: test
+        username: user
```

Comment:
I think the field is called `dn` not `username`, how come it is valid?

Also, please add more test cases, like minLength, dependentRequired. The baseline is, you test every schema you write.